docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -129,13 +140,52 @@ For an end user, the UX to update the kubernetes version of the control plane an
+
+**Notes**:
+- Only compatible changes (as specified in [ClusterClass compatibility](#clusterclass-compatibility)) should be allowed.
+- Changes to InfrastructureMachineTemplates and BootstrapTemplates should be rolled out according to the established operational practices documented in 
+[Updating Machine Infrastructure and Bootstrap Templates](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html), i.e. "template rotation".
+- There are provider-specific incompatible changes which cannot be validated in a "core" webhook, e.g. changing an immutable field of `KubeadmControlPlane`. Those changes will inevitably lead to
+  errors during topology reconciliation. Those errors should be surfaced on the Cluster resource.
+
+#### Story 5 - Ability to customize individual Clusters via variables
+As an end user, I want to customize individual Clusters simply by providing variables in the Cluster resource.
+
+**Note**: Without this ability all Clusters of the same ClusterClass would be the same apart from the properties that are already configure via the topology,
+like Kubernetes version, labels and annotations. This would limit the number of variants a single ClusterClass could address. Variables can e.g. be used to make 
+region/location, machine types and network configuration configurable.
+
+**Example**: A user wants to deploy CAPA Clusters using ClusterClass in different AWS regions. One option to achieve this is to duplicate the ClusterClass and its referenced templates.
+The better option is to introduce a variable and a corresponding patch in the ClusterClass. Now, a user can simply set the AWS region via a variable in the Cluster spec, instead of 
+having to duplicate the entire ClusterClass just to set a different region in the AWSCluster resource.
+
+#### Story 6 - Ability to mutate variables
+As an end user, I want to be able to modify variables in a Cluster. The variable modification should trigger a reconciliation which rolls out the managed topology accordingly, i.e. 
+fields in topology managed resources should be patched.
+
+**Notes**: Same notes as in Story 4 apply.
+
+#### Story 7 - Provide powerful and flexible options to define ClusterClass customizations
+As a ClusterClass author, I want to have powerful and flexible options to specify how templates referenced in a ClusterClass can be customized through patches. These patches should use the user-provided 
+variables defined in a Cluster as input. A ClusterClass author should be able to abstract away details of the provider configuration and thus simplify cluster configuration for the end user.
+
+It should be possible to define simple customizations inline in a ClusterClass without writing any code. To support more advanced use cases, there should be an external mechanism which allows to leverage 
+the full power of a programming language.
```

Comment:
I don't understand "powerful and flexible" in this section and it becomes even more confusing as I read about simplicity and "without writing any code". It's conflicting and confusing.

Perhaps it could be rephrased as a story that states who is trying to do what, maybe? This reads more like a goal to me.

Reply:
I rewrote the whole story and moved it up.